### PR TITLE
Fixed MDS-1170

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/20187

Fixes https://arangodb.atlassian.net/browse/MDS-1170

* Fixed MDS-1170: Significant performance degradation when queries are executed within transactions that involve edits.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.11: https://github.com/arangodb/arangodb/pull/20188
  - [x] Backport for 3.10: this PR

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/MDS-1170
- [ ] Design document:

### CODE REVIEW

**Review by MBkkt — APPROVED** [review]

File: CHANGELOG [hunk 1]
```diff
@@ -19,11 +22,6 @@ v3.10.12 (XXXX-XX-XX)
   connection statistics were built up in memory, but were never released because
   the statistics background thread was not running.
 
-* Avoid memory leak in case an arangod instance is started with the option
```
Comment (neunhoef) [inline CHANGELOG:22]: This does not look right to remove an older CHANGELOG entry. I think this is a merge mistake.

**Review by neunhoef — CHANGES_REQUESTED** [review]
LGTM, but it seems an  older CHANGELOG entry would be lost.

File: CHANGELOG (hunk 1 above)
Comment (jsteemann) [reply CHANGELOG:22]: TBH, I don't know what's going on here. 
In the local version of the branch this CHANGELOG entry is not removed, and my local version is up-to-date. There is nothing left to push.
Maybe this is just a Github UI issue. :shrug:

Comment (max-itzpapalotl) [reply CHANGELOG:22]: OK, let's ignore it.

**Review by neunhoef — APPROVED** [review]
LGTM

**Pull request merged**